onnxsim/onnxsim.cpp

```diff
@@ -444,3 +445,16 @@ onnx::ModelProto Simplify(
   Check(sim_model);
   return sim_model;
 }
+
+void SimplifyPath(const std::string& in_path, const std::string& out_path,
+                  std::optional<std::vector<std::string>> skip_optimizers,
+                  bool constant_folding, bool shape_inference,
+                  size_t tensor_size_threshold) {
+  onnx::ModelProto model;
+  onnx::optimization::loadModel(&model, in_path, true);
+
+  model = Simplify(model, skip_optimizers, constant_folding, shape_inference,
+                   tensor_size_threshold);
```

Comment:
I'm not so familiar with onnx external data. If the weights of `model` are stored as external data, will the constant folding still work? I'm afraid it will raise an error because we have not implemented how to fetch the external data.

Reply:
when `load_external_data` set to true in `onnx::optimization::loadModel` [[src]](https://github.com/onnx/optimizer/blob/master/onnxoptimizer/model_util.cc#L233), it will read from the external file and write the data into `tensor->mutable_raw_data` (see `loadExternalDataForTensor`).

I don't think the optimizer will be affected whether the model is external data or not.

Reply:
Great!

Reply:
BTW, could you join our QQ group (group number: 1021964010, verification answer: nndab) or WeChat group?

![telegram-cloud-photo-size-5-6264841299303706757-y](https://user-images.githubusercontent.com/11607199/180775742-492012bc-b467-4855-a76e-e23dca6d86d3.jpg)

Reply:
yes, data will be storaged in `raw_data` filed  as while as loading into `ModelProto`.